### DISCUSSION TITLE
Updating /modules directory file

### DISCUSSION THREAD

**Discussion by samcoppock** [wintercms/winter#1100] [Q&A]
I have noticed that the composer.json file does not have any reference to the /modules directory?

yet when comparing the changes between v 1.2.1 > 1.2.5 there are many changes to files in this directory.

does updating winter via composer somehow update files in the modules directory.

if yes how?

if no - do these files need to be updated when updating the winter version?

**Comment by jaxwilko** [comment]
Off the top of my head I can't remember where it happens, but anything with the composer `type` property `winter-module` will automatically get moved into `/modules` on install.

This mechanism is the same as the system that moves `winter-plugin` packages into your `plugins` dir on install.

Because of this, generally you would gitignore the `modules` dir and just let composer pull the contents in when you run `composer install`

**Reply by bennothommo** [reply]
It happens as part of the `composer/installers` package that we include with Winter - [this PR](https://github.com/composer/installers/pull/481) added support to Composer to put `winter-module`, `winter-plugin` and `winter-theme` packages in the correct directory.

**Comment by WebVPF** [comment]
```mermaid
graph TB;
    A[modules]-->Backend;
    A-->CMS;
    A-->System;
```

Here they are:
https://github.com/wintercms/winter/blob/65c9cd9cb144bce8d523b5de45da4971706effc0/composer.json#L34-L36

**Comment by LukeTowers** [comment]
@samcoppock wintercms/winter is basically a project template, you can think of it as being similar to laravel/laravel. The code is managed in one place in this repository for simplicity, but it gets split out automatically to the `winter/wn-*-module` repositories / packages which are pulled in via composer. 

You should not track the modules/ folder in your git repositories, just like you should not track the vendor folder, nor any plugins that you include via composer.
